app/services/cycle_timetable.rb

```diff
@@ -33,10 +35,14 @@ def self.current_year
     CYCLE_DATES.keys.detect do |year|
       return year if year == CYCLE_DATES.keys.last
```

Comment:
So this returns the latest year when the next cycle launches right?

i.e. if it's after find reopens it will return 2022?

Sorry if i'm being dense, but how does this differ from just doing

```
return CYCLE_DATES.keys.last
```

Reply:
Have discussed. Me being stupid. Makes sense to me! TIL about detect

Reply:
I wonder if this could be something like 

```
return year if return_year_if_next_cycles_dates_not_set

private

def return_year_if_next_cycles_dates_not_set
  year == CYCLE_DATES.keys.last
end 
```

Just to convey intent

Reply:
Yeah that's a good point. I think there are a few cases in this class where we could extract to method. 
All the methods in this class are `class methods` so we will need to declare private class methods (`private_class_method : next_cycles_dates_not_set?`)